src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,36 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null || user.getLogin() == null || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new RegistrationDataException("User data cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationDataException("User with this login already exists");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationDataException("Login must be at least "
+                    + MIN_LOGIN_LENGTH + " characters");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationDataException("Password must be at least "
+                    + MIN_PASSWORD_LENGTH + " characters");
```

Comment:
Similarly to the login validation, ensure that the password complies with any additional security requirements such as including digits, uppercase letters, or special characters.